frontend.md

```diff
@@ -12,20 +12,65 @@ A few general rules for frontend:
-    - A **`API key`**: This is a key retrieved from Hypixel and provided by the developer, this should be in the form of a **UUID**.
-    - A (reference to a) **`HypixelCache`**: This is a data structure that implements the `HypixelCache` interface, see below.
-    - A (reference to a) **`HypixelCacheStrategy`**: This is a data structure that implements the `HypixelCacheStrategy` interface, see below.</br>
- - **`HypixelAPI`** **must** implement the following methods (for the different data structures, see below):
-    - A method that returns a **`HypixelPlayer`** data structure. This method takes either a UUID or the name of the requested player as argument.
-    - A method that returns a **`HypixelGuild`** data structure. This method takes either the UUID of a player in the requested guild, the id of the requested guild or the name of the requested guild as argument.
-    - A method that returns a **`HypixelLeaderboards`** data structure. This method takes no arguments.
-    - A method that returns a **`HypixelPunishmentStats`** data structure. This method takes no arguments.</br>
- - **`HypixelPlayer`** **must** implement the following methods:
-    - A method that returns the friends of the player in the form of a list of **`HypixelPlayer`** data structures.
-    - A method that returns the online status of the player in the form of a **`HypixelPlayerStatus`** data structure.
-    - A method that returns the recent games of the player in the form of a list of **`HypixelGameSession`** data structures.
-    - A method that returns either a **`HypixelGuild`** if the player is in a guild or nothing if the player is not in a guild.</br>
+**Note**: When reading the information below, multiple methods that return the same data **must** only ever be allowed because of multiple independent arguments.
+
+**Note**: This information explicitly does not describe how the different data structures should behave internally or what data they need to store. This is reserved for the backend and should not concern the frontend enough to be listed here.
+
+- A main data structure called `HypixelAPI` **must** exist in the wrapper. `HypixelAPI` **must** store the following data:
+
+| name | type | note |
+| --- | --- | --- |
+| API Key | **`UUID`** | This is a key retrieved from Hypixel and provided by the developer. |
+| Cache | **`HypixelCache`** | This may be a reference. `HypixelCache` is an interface, see below. |
+| Cache Strategy | **`HypixelCacheStrategy`** | This may be a reference. `HypixelCacheStrategy` is an interface, see below. |
+
+- **`HypixelAPI` must** implement one or more methods for every row in the following table:
+
+| arguments | return type | note |
+| --- | --- | --- |
+| or `UUID` of the player</br>or `Name` of the player | **`HypixelPlayer`** | `Name` is a string. |
+| or `UUID` of player in guild</br>or `ID` of requested guild</br>or `Name` of the requested guild | **`HypixelGuild`** | `Name` is a string. |
```

Comment:
At the moment, getting a player's guild is planned to be done off of `HypixelPlayer` itself (e.g. `HypixelPlayer.getGuild()`), rather than off of `HypixelAPI`. By name & id are fine here though
```suggestion
| or `ID` of requested guild</br>or `Name` of the requested guild | **`HypixelGuild`** | `Name` is a string. |
```